RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -851,51 +991,28 @@ void PFAlgo::elementLoop(const reco::PFBlock& block,
                          ElementIndices& inds,
                          std::vector<bool>& deadArea) {
   LogTrace("PFAlgo|elementLoop") << "start of function PFAlgo::elementLoop, elements.size()" << elements.size();
+
   for (unsigned iEle = 0; iEle < elements.size(); iEle++) {
     PFBlockElement::Type type = elements[iEle].type();
 
     LogTrace("PFAlgo|elementLoop") << "elements[iEle=" << iEle << "]=" << elements[iEle];
+    //only process TRACK elements, but fill the ElementIndices vector with indices for all elements.
+    //Mark the active & deadArea for bad HCAL
     auto ret_decideType = decideType(elements, type, active, inds, deadArea, iEle);
     if (ret_decideType == 1) {
       LogTrace("PFAlgo|elementLoop") << "ret_decideType==1, continuing";
       continue;
     }
-    LogTrace("PFAlgo|elementLoop") << "ret_decideType=" << ret_decideType;
-
-    // we're now dealing with a track
-    unsigned iTrack = iEle;
-    reco::MuonRef muonRef = elements[iTrack].muonRef();
-    if (muonRef.isNonnull()) {
-      LogTrace("PFAlgo|elementLoop") << "track " << iTrack << " has a valid muon reference. " << endl
```

Comment:
I do not see a replacement for this check on muonRef.
Is this just an unnecessary debugging piece?

Reply:
Thanks for spotting that. Since `isNonnull` does not throw, in essence, this is just a debugging piece. I removed it since `muonRef` is not used anywhere else in function, so we might as well keep the code simpler.